repair/repair.cc

```diff
@@ -266,6 +266,7 @@ void tracker::done(int id, bool succeeded) {
     } else {
         _status[id] = repair_status::FAILED;
     }
+    _done_cond.broadcast();
```

Comment:
Will this get triggered on shutdown?

Please test shutdown with this call running, and a repair running, in debug mode, to check for use-after-free.

Reply:
(and to make sure we return to the caller with a reasonable response).

Reply:
Yes. When shutdown happens, repair jobs will be aborted and tracker::done() will be called.

The caller will get a response saying the repair failed.